nautobot/dcim/tests/test_api.py

```diff
@@ -1770,6 +1770,30 @@ def setUpTestData(cls):
             "domain": "newdomain",
         }
 
+    def test_master_device_vc_assignment(self):
+        """Test device assigned vc_position keeps assignment after being set to master.
+
+        This test is for https://github.com/nautobot/nautobot/issues/393
+        """
+        device = Device.objects.get(name="Device 12")
+        device.vc_position = 3
+        device.save()
+        self.assertEquals(device.vc_position, 3)
+        VirtualChassis.objects.create(name="Virtual Chassis 1", master=device, domain="domain-1")
+        device = Device.objects.get(name="Device 12")
```

Comment:
Consider instead:

```suggestion
        device.refresh_from_db()
```

Same suggestion below.